interactor-impl/src/main/java/com/wednesday/template/interactor/localFm/search/SearchAlbumInteractorImpl.kt

```diff
@@ -2,34 +2,60 @@ package com.wednesday.template.interactor.localFm.search
 
 import com.wednesday.template.domain.base.Result
 import com.wednesday.template.domain.lastFm.Album
+import com.wednesday.template.domain.lastFm.GetFavouriteAlbumFlowUseCase
 import com.wednesday.template.domain.lastFm.SearchAlbumUseCase
 import com.wednesday.template.interactor.base.CoroutineContextController
 import com.wednesday.template.interactor.localFm.SearchAlbumInteractor
 import com.wednesday.template.presentation.base.UIList
+import com.wednesday.template.presentation.base.UIResult
 import kotlinx.coroutines.channels.Channel
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.consumeAsFlow
+import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.flowOn
-import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.receiveAsFlow
 
 class SearchAlbumInteractorImpl(
     private val searchAlbumUseCase: SearchAlbumUseCase,
     private val coroutineContextController: CoroutineContextController,
-    private val uiAlbumMapper: UIAlbumMapper
+    private val uiAlbumSearchMapper: UIAlbumSearchMapper,
+    favouriteAlbumFlowUseCase: GetFavouriteAlbumFlowUseCase
 ) : SearchAlbumInteractor {
 
     private val searchResultChannel = Channel<List<Album>> { }
 
-    override val searchResultsFlow: Flow<UIList> = searchResultChannel
-        .consumeAsFlow()
-        .map {
-            UIList(uiAlbumMapperImpl.map(it))
-        }
-        .flowOn(coroutineContextController.dispatcherDefault)
+//    override val searchResultsFlow: Flow<UIList> = searchResultChannel
+//        .consumeAsFlow()
+//        .map {
+//            UIList(uiAlbumMapper.map(it,false))
+//        }
+//        .flowOn(coroutineContextController.dispatcherDefault)
 //        .catch { e ->
 //            emit(UIResult.Error(e as Exception ))
 //        }
 
+    override val searchResultsFlow: Flow<UIResult<UIList>> = favouriteAlbumFlowUseCase(Unit)
+        .combine(searchResultChannel.receiveAsFlow()) { favouriteAlbum, searchResult ->
+            when {
+                searchResult.isEmpty() -> {
+                    UIResult.Success(
+                        UIList()
+                    )
+                }
+                favouriteAlbum is Result.Success -> {
+                    UIResult.Success(
+                        uiAlbumSearchMapper.map(favouriteAlbum.data, searchResult)
+                    )
+                }
+                favouriteAlbum is Result.Error -> {
+                    UIResult.Error(favouriteAlbum.exception)
+                }
+                else -> {
+                    UIResult.Error(error("Something went wrong"))
```

Comment:
The `error` function is an inbuilt function from kotlin. It's used to easily throw errors. I think it will not return an error to use but instead throw the error. Which is not what we want here.

Reply:
implemented the required change.